apps/studio/src/features/editing-experience/components/ChildPageEditorDrawer.tsx

```diff
@@ -0,0 +1,183 @@
+  IsomerSchema,
+} from "@opengovsg/isomer-components"
+import type { Static } from "@sinclair/typebox"
+import { useCallback } from "react"
+import { Box, Flex, useDisclosure } from "@chakra-ui/react"
+import { Button, useToast } from "@opengovsg/design-system-react"
+import { getChildpageSchema } from "@opengovsg/isomer-components"
+import Ajv from "ajv"
+import isEmpty from "lodash/isEmpty"
+import isEqual from "lodash/isEqual"
+
+import { BRIEF_TOAST_SETTINGS } from "~/constants/toast"
+import { useEditorDrawerContext } from "~/contexts/EditorDrawerContext"
+import { useQueryParse } from "~/hooks/useQueryParse"
+import { trpc } from "~/utils/trpc"
+import { editPageSchema } from "../schema"
+import { CHANGES_SAVED_PLEASE_PUBLISH_MESSAGE } from "./constants"
+import { DiscardChangesModal } from "./DiscardChangesModal"
+import { DrawerHeader } from "./Drawer/DrawerHeader"
+import { ErrorProvider, useBuilderErrors } from "./form-builder/ErrorProvider"
+import FormBuilder from "./form-builder/FormBuilder"
+
+const ajv = new Ajv({ strict: false, logger: false })
+
+// NOTE: by the time we display this drawer,
+// the only allowed `layout` type shold be `index`.
+// This invariant is upheld in `RootStateDrawer`,
+// where we only show the fixed block that triggers this drawer
+// when the `layout` of the `page` is `index`
+const narrowToIndex = (
+  page: Omit<IsomerSchema, "version">,
+): page is Static<typeof IndexPageSchema> => {
+  return page.layout === "index"
+}
+
+export default function ChildPageEditorDrawer(): JSX.Element {
+  const {
+    isOpen: isDiscardChangesModalOpen,
+    onOpen: onDiscardChangesModalOpen,
+    onClose: onDiscardChangesModalClose,
+  } = useDisclosure()
+  const {
+    setDrawerState,
+    savedPageState,
+    setSavedPageState,
+    previewPageState,
+    currActiveIdx,
+    setPreviewPageState,
+  } = useEditorDrawerContext()
+
+  if (!narrowToIndex(previewPageState)) {
+    throw new Error()
+  }
+
+  const toast = useToast()
+
+  const schema = getChildpageSchema()
+  const validateFn = ajv.compile<Static<typeof schema>>(schema)
+
+  const { pageId, siteId } = useQueryParse(editPageSchema)
+  const utils = trpc.useUtils()
+  const { mutate, isLoading } = trpc.page.updatePageBlob.useMutation({
+    onSuccess: async () => {
+      await utils.page.readPageAndBlob.invalidate({ pageId, siteId })
+      await utils.page.readPage.invalidate({ pageId, siteId })
+      toast({
+        title: CHANGES_SAVED_PLEASE_PUBLISH_MESSAGE,
+        ...BRIEF_TOAST_SETTINGS,
+      })
+    },
+  })
+
+  const handleSaveChanges = useCallback(() => {
+    mutate(
+      {
+        pageId,
+        siteId,
+        content: JSON.stringify(previewPageState),
+      },
+      {
+        onSuccess: () => setDrawerState({ state: "root" }),
+      },
+    )
+  }, [
+    currActiveIdx,
+    mutate,
+    pageId,
+    previewPageState,
+    setDrawerState,
+    setPreviewPageState,
+    setSavedPageState,
+    siteId,
+    toast,
+  ])
```

Comment:
remove unnecessary dependencies? 
`currActiveIdx`, `setPreviewPageState`, `setSavedPageState`, `toast`

Reply:
thanks! btw, does your linter warn you here? i had no diagnostic

Reply:
yeap it shows up in my eslint extension eh haha